src/multiformats/address.cljc

```diff
@@ -123,16 +119,19 @@
         value (when (coll? entry) (second entry))
         {:keys [code, codec]} (ensure-protocol-attrs protocol-key)
         code-bytes (varint/encode code)
-        val-bytes (if (nil? value)
-                    (b/byte-array 0)
+        fixed-len (codec/fixed-byte-length codec)
+        ;; no value bytes needed for no-value protocols (fixed-length at 0)
+        val-bytes (when (or (not fixed-len) (pos? fixed-len))
                     (codec/str->bytes codec value))
-        val-len-bytes (if (codec/fixed-byte-length codec)
-                        (b/byte-array 0)
+        ;; only have length bytes for variable length protocols
+        val-len-bytes (when-not fixed-len
                         (varint/encode (alength ^bytes val-bytes)))]
-    (concat-arrs code-bytes val-len-bytes val-bytes)))
+    (->> [code-bytes val-len-bytes val-bytes]
+         (remove nil?)
+         (apply concat-arrs))))
```

Comment:
I think I'll update alphabase so that this function ignores nils, which would make this form simpler:
```clojure
(b/concat code-bytes val-len-bytes val-bytes)
```